### DISCUSSION TITLE
Questions about the user object

### DISCUSSION THREAD

**Discussion by blacktime1999** [EchterAlsFake/unofficial-api-for-pornhub#25] [Q&A] [answered]
![image](https://github.com/Egsagon/PHUB/assets/95710926/9a63389b-cbed-4170-8b70-73d26b7f931a)
'get_user()' acquired user object ,Why only one attribute

**Comment by blacktime1999** [comment]
![image](https://github.com/Egsagon/PHUB/assets/95710926/d7577aed-1dcf-405a-a4d7-e9d1eec82b1c)
There should be so many attributes.
Am I understanding this correctly?

**Comment by Egsagon** [comment, marked as answer]
Hi,
As stated in the documentation, the output of `client.get_user` is a `User` object.
When printed in the console, only the name of the user is printed for visibility purposes.

In fact, at this point in your code none of the data is actually fetched. This is for optimization purposes. If you want to access data, you need to call it:

```py
user = client.get_user('...')

print(user.url)
print(user.info)
print(user.videos)
# etc.
```

Note each of these values will be cached once you call them. This means that their value won't refresh, even if the real pornhub values change. You can refresh them using `user.refresh()`.

This behavior is implemented on all PHUB dataclasses (user, video, image, etc.)

**Reply by blacktime1999** [reply]
不行呀

**Reply by blacktime1999** [reply]
![image](https://github.com/Egsagon/PHUB/assets/95710926/ce774a2a-d3a7-4ea6-b53e-2018377ffea9)
这样的返回值，是有问题的吧
应该返回一个list，可是它没有

**Reply by Egsagon** [reply]
The project structure might look a little bit complicated.

`user_info.videos` cannot be a list. Imagine if a user has hundreds of videos, the output list would be really long and take a lot of space in your memory. Instead, we are using a [query](https://phub.readthedocs.io/en/latest/api/objects.html#phub.objects.Query) object. This object automatically manages what requests are being sent to the servers, and what data is cached.

This object is used for a lot of purposes in PHUB, such as [searching](https://phub.readthedocs.io/en/latest/api/objects.html#phub.objects.Query), or in your case, enumerating user videos.

You typically want to use query objects like so:

```py
client = phub.Client()
user = client.get_user(...)

for video in user.videos:
    print(video)
```

The above exemple will go through every video of the user. If you are worried this sends too much requests to Pornhub servers, you can use python slices syntax to enumerate only a chunk of these videos. For exemple:
```py
# Enumerate the 10 most recent videos
for video in user.videos[0:10]:
    print(video)
```

**Reply by blacktime1999** [reply]
我知道

**Reply by blacktime1999** [reply]
but

**Comment by blacktime1999** [comment]
![image](https://github.com/Egsagon/PHUB/assets/95710926/1421ba58-9811-4275-b42b-0764d398f490)
不知道你运行这样的代码是什么返回值，给我的返回值不是视频的url，而是博主的url

**Reply by Egsagon** [reply]
You made a typo on line 6.
You were taking the url attribute from the fetched user and not from the query result.

```py
client = phub.Client()
user = client.get_user('asmodeo_69')
for video in user.videos[0:5]:
    print(video)
```
The output should be:
```
phub.Video(key=6532c78f24b84)
phub.Video(key=6515e0ee048da)
phub.Video(key=6505206963952)
phub.Video(key=64ffab8e9272a)
phub.Video(key=64dfc5509ea25)
```

Once again, each of these video are `Video` objects. You can find more documentation about them [here](https://phub.readthedocs.io/en/latest/features/video.html).

**Reply by blacktime1999** [reply]
你真的18岁？

**Reply by blacktime1999** [reply]
Thank you very much.

**Reply by Egsagon** [reply]
Yep, im 18.
Glad i could help you
